### ISSUE TITLE
Invalid DOI should not show "Connection error"

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#8127]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Open https://link.springer.com/book/10.1007/978-3-319-47590-5
2. Download PDF ("Download book PDF")
3. Go to page 207
4. Select and copy the DOI:  
  ![image](https://user-images.githubusercontent.com/1366654/136320107-717ffb77-a6e2-4c37-86da-2993ccd2dd07.png)
5. Open "New entry dialog" and paste DOI:  
  ![image](https://user-images.githubusercontent.com/1366654/136320161-9e41e557-8cc0-408e-bc59-28621f5b9734.png)
6. See the question mark at the end
7. Click "Generate"
8. Result:  
  ![image](https://user-images.githubusercontent.com/1366654/136320213-9bfe6a19-9679-4bca-80c1-7c30006ec2b8.png)

Expected behavior:

- The invalid character at the end should be removed internally
- The DOI data should be fetched

**Comment by TheDeveloperDino** [comment]
Hello, is this issue free? I can work on this issue.

**Comment by koppor** [comment]
@TheDeveloperDino You can to test-driven development. A start can be https://github.com/JabRef/jabref/pull/8124. I think, one should add test cases to `org.jabref.model.entry.identifier.DOITest` and then work on `org.jabref.model.entry.identifier.DOI`.

**Comment by mrcstan** [comment]
@TheDeveloperDino, are you still working on this? @koppor , I don't use Windows (I use Mac) but I would like to fix it by removing any trailing invalid characters at the end of the DOI.

**Comment by jiezheng5** [comment]
Hi, 

I am a cs student, I can repeat the issue in both Windows 10 and Linux Ubuntn OS 20.04. Can I work on this issue?

**Comment by mrcstan** [comment]
@jiezheng5, this is not really in the spirit of open source contribution. You requested to work on this issue after me and we have not heard from @TheDeveloperDino

**Comment by jiezheng5** [comment]
> @jiezheng5, this is not really in the spirit of open source contribution. You requested to work on this issue after me and we have not heard from @TheDeveloperDino. I have also started working on the issue.

I actually started working on this issue 2 weeks ago. Then I saw your request, I thought maybe I should also post my progress. Do we have to first get permission before we can work on the open issues? This is my first time contributing to open source, I am sorry that if I have done something wrong.

**Comment by koppor** [comment]
Example DOI for this behavior: `10.3218/3846-0`. It is a valid DOI string - but not not found on doi org: https://doi.org/10.3218/3846-0 returns 404.

When a user fetches the DOI data, "Connection error" is returned. The improved implementation should check for `404` and handle that as "No DOI data existing".

Side note: The DOI is listed at https://vdf.ch/studienarbeiten-e-book.html - and thus users may think, it is a valid one. However, the publisher seems not to have registered it at the official DOI registry.

**Comment by koppor** [comment]
> I fixed this issue by adding a method to automatically remove non-valid characters at the end of DOI string

@mrcstan Could you add a test case to `org.jabref.model.entry.identifier.DOITest`? I think, the solution is found by working on `org.jabref.model.entry.identifier.DOI#DOI_EXP`, which seems to include more characters than needed in the resulting DOI?

**Comment by koppor** [comment]
> Then I saw your request, I thought maybe I should also post my progress. 

This is actually a good idea. We also outlined that at https://devdocs.jabref.org/contributing#create-a-pull-request:

>> If you want to indicate that a pull request is not yet complete before creating the pull request, you may consider creating a draft pull request.

> Do we have to first get permission before we can work on the open issues? 

No, just start working. The general reason is that the probability at 500+ opened issues that two volunteers start working on the same issue at the same time is nearly 0. Maybe in the concrete setting, this issue was a) tagged as "good first issue" and b) at the very beginning of the list.

I am sorry that we have this situation now. I reviewed https://github.com/JabRef/jabref/pull/8215/ and think, it would be good to see an alternative solution.

Proposal:

1. @mrcstan Submits his solution as WIP PR
2. The JabRef developers compare the two solutions and decide which one is closer to a solution based on our 15+ years experience in Java development

Would that be OK?

> This is my first time contributing to open source,

Welcome to the open source development world!

> I am sorry that if I have done something wrong.

You "just" hit an edge case in the open source world. We are now working to resolve this case. Meanwhile, you could work on another issue, maybe https://github.com/koppor/jabref/issues/534?

**Comment by koppor** [comment]
This PR has two parts:

1. Invalid characters at a DOI --> fixed by https://github.com/JabRef/jabref/pull/8227
2. DOI is a valid one, but not returned by the server --> still open

**Comment by mrcstan** [comment]
@koppor , i just submitted a PR #8127 addressing 1 and 2. I see that you have address 1. Let me know your thoughts.
